internal/operands/metrics/resources.go

```diff
@@ -61,31 +72,87 @@ func ServiceMonitorLabels() map[string]string {
 	}
 }
 
-func newServiceMonitorCR(namespace string) *promv1.ServiceMonitor {
-	return &promv1.ServiceMonitor{
+func serviceCABundle() promv1.SecretOrConfigMap {
+	return promv1.SecretOrConfigMap{
+		ConfigMap: &v1.ConfigMapKeySelector{
+			LocalObjectReference: v1.LocalObjectReference{
+				Name: ServiceCABundle,
+			},
+			Key: ServiceCABUndleKey,
+		},
+	}
+}
+
+func olmManagedCABundle() promv1.SecretOrConfigMap {
+	return promv1.SecretOrConfigMap{
+		Secret: &v1.SecretKeySelector{
+			LocalObjectReference: v1.LocalObjectReference{
+				Name: OLMManagedCert,
+			},
+			Key: OLMManagedCertKey,
+		},
+	}
+}
+
+func getCAConfigForServiceMonitor(olmDeployment bool) promv1.SecretOrConfigMap {
+	if olmDeployment {
+		return olmManagedCABundle()
+	}
+	return serviceCABundle()
+}
+
+func newValidatorServiceMonitor(request common.Request) *promv1.ServiceMonitor {
+	tlsConfig := &promv1.TLSConfig{
+		SafeTLSConfig: promv1.SafeTLSConfig{
+			CA: serviceCABundle(),
+		},
+	}
+	tlsConfig.ServerName = ptr.To(fmt.Sprintf("virt-template-validator.%s.svc", request.Namespace))
+
+	serviceMonitor := newServiceMonitor(TemplateValidatorMetricsServiceName, request.Namespace, tlsConfig, metav1.LabelSelector{
+		MatchLabels: map[string]string{
+			MetricsServiceKey: TemplateValidatorMetricsServiceName,
+		},
+	})
+	return &serviceMonitor
+}
+
+func newSspServiceMonitor(request common.Request) *promv1.ServiceMonitor {
```

Comment:
Similarly here, can you only pass required fields to the function instead of whole `request` struct?